### ISSUE TITLE
macos binary - quits on startup

### ISSUE THREAD

**Issue by martinleach** [nanoporetech/dorado#33]
downloaded the binary for a mac (chipset: Apple M1 Max) - OS: Monterery 12.5.1

I get this error when calling dorado from command-line

dyld[7721]: Library not loaded: '/opt/homebrew/opt/libaec/lib/libsz.2.dylib'

any suggestions?

**Comment by martinleach** [comment]
after digging

brew install hdf5

added in the necessary libraries and allowed dorado to run
suggest including these libraries or adding a pre-req to install/update hdf5

**Comment by vellamike** [comment]
@martinleach  thank you for identifying this issue and reporting your solution. We are currently working on making Dorado installation more smooth so this is very useful. I will keep this issue open until this has been resolved.

**Comment by vdejager** [comment]
thanks @martinleach , this was very helpful indeed.

**Comment by photocyte** [comment]
Hi, I also ran into a very similar issue. 

```
dyld[54914]: Library not loaded: '/opt/homebrew/opt/libaec/lib/libsz.2.dylib'
  Referenced from: '/Users/tim/Downloads/dorado-0.1.0+4b0e9a6-Darwin/bin/dorado'
  Reason: tried: '/opt/homebrew/opt/libaec/lib/libsz.2.dylib' (no such file), '/usr/local/lib/libsz.2.dylib' (no such file), '/usr/lib/libsz.2.dylib' (no such file)
```

I don't use homebrew, I use macports. I tried `sudo port install hdf5` and that didn't seem to fix it. I then tried `sudo port install libaec` , and the symlinked into my native paths:

```
sudo ln -s /opt/local/lib/libaec/lib/libsz* /usr/local/lib/ 
``` 

I think that worked. Then, I got a new error:
```
dyld[59444]: Library not loaded: '/opt/homebrew/opt/zstd/lib/libzstd.1.dylib'
  Referenced from: '/Users/tim/Downloads/dorado-0.1.0+4b0e9a6-Darwin/bin/dorado'
  Reason: tried: '/opt/homebrew/opt/zstd/lib/libzstd.1.dylib' (no such file), '/usr/local/lib/libzstd.1.dylib' (no such file), '/usr/lib/libzstd.1.dylib' (no such file)
zsh: abort      /Users/tim/Downloads/dorado-0.1.0+4b0e9a6-Darwin/bin/dorado
```

So, `sudo port install zstd`

```
sudo ln -s /opt/local/lib/libzstd* /usr/local/lib/
```

```
dyld[59545]: Library not loaded: '/opt/homebrew/opt/openssl@3/lib/libssl.3.dylib'
  Referenced from: '/Users/tim/Downloads/dorado-0.1.0+4b0e9a6-Darwin/bin/dorado'
  Reason: tried: '/opt/homebrew/opt/openssl@3/lib/libssl.3.dylib' (no such file), '/usr/local/lib/libssl.3.dylib' (no such file), '/usr/lib/libssl.3.dylib' (no such file)
zsh: abort      /Users/tim/Downloads/dorado-0.1.0+4b0e9a6-Darwin/bin/dorado
```

So... `sudo port install openssl`

```
sudo ln -s /opt/local/lib/openssl-3/libssl* /usr/local/lib/
```

```
dyld[59676]: Library not loaded: '/opt/homebrew/opt/openssl@3/lib/libcrypto.3.dylib'
  Referenced from: '/Users/tim/Downloads/dorado-0.1.0+4b0e9a6-Darwin/bin/dorado'
  Reason: tried: '/opt/homebrew/opt/openssl@3/lib/libcrypto.3.dylib' (no such file), '/usr/local/lib/libcrypto.3.dylib' (no such file), '/usr/lib/libcrypto.3.dylib' (no such file)
zsh: abort      /Users/tim/Downloads/dorado-0.1.0+4b0e9a6-Darwin/bin/dorado
```

I stopped there. Could you please support Macports (https://www.macports.org) installed dependencies in Dorado, as well as homebrew installations?

**Comment by tnn111** [comment]
Are there installation instructions for the Mac anywhere? I get the following error:

dyld[18120]: Library not loaded: /opt/homebrew/opt/libaec/lib/libsz.2.dylib
  Referenced from: <FA4C1798-1986-3762-8A87-76398448D877> /Users/torben/opt/dorado-0.1.1/bin/dorado
  Reason: tried: '/opt/homebrew/opt/libaec/lib/libsz.2.dylib' (no such file), '/System/Volumes/Preboot/Cryptexes/OS/opt/homebrew/opt/libaec/lib/libsz.2.dylib' (no such file), '/opt/homebrew/opt/libaec/lib/libsz.2.dylib' (no such file), '/usr/local/lib/libsz.2.dylib' (no such file), '/usr/lib/libsz.2.dylib' (no such file, not in dyld cache)
zsh: abort      ./dorado -h

I suppose I can go and play with symlinks as well, but I'd rather not if it can be avoided. Has anyone played with mamba/conda for this? 

Thanks, Torben

**Comment by tijyojwad** [comment]
The missing libraries are now shipped with the Mac package starting from v0.3.1 - https://cdn.oxfordnanoportal.com/software/analysis/dorado-0.3.1-osx-arm64.tar.gz
